### PR TITLE
Show icons in redesigned and existing progress dots (Unrevert 13212)

### PR DESCRIPTION
**Pull request by Bjvanminnen**
#13212 was reverted because when our progressRedesign experiment was not enabled, assessment level were not showing up properly. This is because even tho`iconClassName` knew to ignore 'fa-list-ol', we also rendered differently based on whether or not we had a `level.icon`.

This unifies the logic a little bit more, and gets us closer to the place I'd like us to be where the server sends us an IconType instead of a className. I'm essentially pretending that's what it does, except that the IconType key is actually just the className (if that makes sense)

### CODE REVIEW

File: dashboard/app/models/script_level.rb [hunk 1]
```diff
@@ -177,6 +177,14 @@ def path
     build_script_level_path(self)
   end
 
+  def icon
+    if level.icon
+      level.icon
+    elsif assessment
+      'fa-list-ol'
+    end
+  end
```
Comment (Bjvanminnen) [inline dashboard/app/models/script_level.rb]: @joshlory FYI for overriding icon for assessments. Looking at this again, I think that this (a) deserves a comment that captures some of the discussion we've had around icons and (b) I probably should reverse the conditionals, i.e.
```
if assessment
  'fa-list-ol'
else
  level.icon
end
```
Even though there probably aren't currently any level types that have an icon, that we would also make an assessment, if some existing type later did provide a default icon, I think we'd still want assessment to override.

File: apps/src/code-studio/components/progress/ProgressDot.jsx [hunk 2]
```diff
@@ -134,6 +134,19 @@ const styles = {
...
+// className. For now, I'm going to use the className in level.icon as if it
+// were actually a type key.
+const iconClassFromIconType = {
+  'fa-file-text': 'fa fa-file-text',
+  // Explicitly don't want to use an icon for this type
+  'fa-list-ol': undefined,
+  'fa-external-link-square': 'fa fa-external-link-square',
+  'fa-video-camera': 'fa fa-video-camera',
+  'fa-stop-circle': 'fa fa-stop-circle',
+  'fa-map': 'fa fa-map',
```
Comment (islemaster) [inline apps/src/code-studio/components/progress/ProgressDot.jsx:147]: :+1:

File: apps/src/code-studio/components/progress/ProgressDot.jsx [hunk 3]
```diff
@@ -202,8 +215,8 @@ export const ProgressDot = Radium(React.createClass({
 
     // fa-list-ol is used only by our redesigned dots, but we don't want to use
     // it here
```
Comment (islemaster) [inline apps/src/code-studio/components/progress/ProgressDot.jsx]: The `fa-list-ol` special case comment here should maybe be moved up to your mapping.

File: apps/src/code-studio/components/progress/ProgressDot.jsx [hunk 4]
```diff
@@ -202,8 +215,8 @@ export const ProgressDot = Radium(React.createClass({
...
-    if (level.icon && level.icon !== 'fa-list-ol') {
-      return 'fa ' + level.icon;
+    if (level.icon) {
+      return iconClassFromIconType[level.icon];
```
Comment (islemaster) [inline apps/src/code-studio/components/progress/ProgressDot.jsx]: You used to return `''` empty string in the `fa-list-ol` case; now you return `undefined`.  Intentional?

File: apps/test/unit/code-studio/components/progress/ProgressDotTest.js [hunk 5]
```diff
@@ -443,7 +443,8 @@ describe('ProgressDot component tests', () => {
         );
 
         const result = renderer.getRenderOutput();
-        expect(result.props.children[0].props.className).to.equal('');
+        expect(result.props.children[0].type).to.equal('div');
+        expect(result.props.children[0].props.className).to.equal(undefined);
```
Comment (islemaster) [inline apps/test/unit/code-studio/components/progress/ProgressDotTest.js:447]: Ah, seems intentional.

Comment (Bjvanminnen) [reply apps/test/unit/code-studio/components/progress/ProgressDotTest.js:447]: Don't think it makes a big difference one way or the other really, but `undefined` seems slightly more accurate.

Comment (islemaster) [reply apps/test/unit/code-studio/components/progress/ProgressDotTest.js:447]: Should your default case return `undefined` then?

Comment (Bjvanminnen) [reply apps/test/unit/code-studio/components/progress/ProgressDotTest.js:447]: Ahh yes, probably. I expect to do some future work here in the near future, so I'll try to remember to clean that up then.

**Pull request merged**